securedrop_export/export.py

```diff
@@ -277,31 +278,42 @@ def unlock_luks_volume(self, encryption_key):
                 if 'UUID' in items[0]:
                     self.encrypted_device = 'luks-' + items[1]
 
-            # the luks device is not already unlocked
-            if not os.path.exists(os.path.join("/dev/mapper/", self.encrypted_device)):
-                logger.debug('Unlocking luks volume {}'.format(self.encrypted_device))
-                p = subprocess.Popen(
-                    ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
-                    stdin=subprocess.PIPE,
-                    stdout=subprocess.PIPE,
-                    stderr=subprocess.PIPE
-                )
-                logger.debug('Passing key')
-                p.communicate(input=str.encode(encryption_key, "utf-8"))
-                rc = p.returncode
-                if rc != 0:
-                    logger.error('Bad phassphrase for {}'.format(self.encrypted_device))
-                    self.exit_gracefully(ExportStatus.USB_BAD_PASSPHRASE.value)
+            # the luks device is already unlocked
+            if os.path.exists(os.path.join('/dev/mapper/', self.encrypted_device)):
+                logger.debug('Device already unlocked')
+                return
```

Comment:
Just confirming I'm not missing something here: the `unlock_luks_volume` method is functionally equivalent to before? Since it will only run the logic beginning with `logger.debug('Unlocking luks volume {}'.format(self.encrypted_device))` if not `os.path.exists(os.path.join('/dev/mapper/', self.encrypted_device))`?

Reply:
correct, instead of putting the expected case in the if block, i put the unusual case there